spinoso-string/src/codepoints.rs

```diff
@@ -125,7 +193,7 @@ impl<'a> TryFrom<&'a String> for Codepoints<'a> {
     fn try_from(s: &'a String) -> Result<Self, Self::Error> {
         let state = match s.encoding() {
             Encoding::Utf8 => {
-                if let Ok(s) = s.buf.to_str() {
+                if let Ok(s) = s.inner.as_vec().to_str() {
```

Comment:
`to_str` is `ByteSlice::to_str`. As long as `ByteSlice` is in scope, this method is available on `&[u8]`.

```suggestion
                if let Ok(s) = s.inner.as_slice().to_str() {
```